specification/security/resource-manager/Microsoft.Security/preview/2024-07-01-preview/trustedIps.json

```diff
@@ -0,0 +1,323 @@
+      }
+    }
+  },
+  "paths": {
+    "/{scope}/providers/Microsoft.Security/trustedIps": {
+      "get": {
+        "x-ms-examples": {
+          "List trusted IPs by subscription scope": {
+            "$ref": "./examples/trustedIps/ListBySubscriptionTrustedIps_example.json"
+          }
+        },
+        "tags": [
+          "TrustedIps"
+        ],
+        "description": "Get a list of all relevant trusted IPs over a scope",
+        "operationId": "TrustedIps_List",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/Scope"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/TrustedIpsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Security/trustedIps/{trustedIpsId}": {
+      "get": {
+        "x-ms-examples": {
+          "Get a trusted IPs object over security connector scope": {
+            "$ref": "./examples/trustedIps/GetSecurityConnectorTrustedIps_example.json"
+          }
+        },
+        "tags": [
+          "TrustedIps"
+        ],
+        "description": "Get a specific trusted IPs object for the requested scope by trusted IPs Id",
+        "operationId": "TrustedIps_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/Scope"
+          },
+          {
+            "$ref": "#/parameters/TrustedIpsId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/TrustedIps"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Create or update trusted IPs over subscription scope": {
+            "$ref": "./examples/trustedIps/PutSubscriptionTrustedIps_example.json"
+          }
+        },
+        "tags": [
+          "TrustedIps"
+        ],
+        "description": "Creates or updates a trusted IPs object over a given scope",
+        "operationId": "TrustedIps_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
```

Comment:
>ApiVersion [](http://example.com/codeflow?start=63&length=10)

There are definitions of common parameters that you can use instead of declaring your own.

Reply:
We did it this way because of the uniformity with the rest of our ARM models.
This will be refactored in the next version of the API